src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/CompareUtil.java

```diff
@@ -122,4 +125,41 @@ public char getOperator() {
             return '~';
         }
     }
+
+    /**
+     * Compares just like plain comparison, however, patterns are expanded using the
+     * given system environment settings.
+     */
+    class PlainVarCompareUtil implements CompareUtil {
```

Comment:
I think it would be better to skip this one and add the functionality to the existing Plain comparer, deprecate the old `matches` method and introduce a new one that also takes an EnvVars as parameter.